shelley/chain-and-ledger/formal-spec/update.tex

```diff
@@ -0,0 +1,396 @@
+      \begin{array}{l}
+        \var{slot}\\
+        \var{dms}\\
+      \end{array}
+      \vdash
+      \left(
+      \begin{array}{l}
+        \var{aup_s}\\
+        \var{favs}\\
+        \var{avs}\\
+      \end{array}
+      \right)
+      \trans{avup}{aup}
+      \left(
+      \begin{array}{l}
+        \varUpdate{\var{aup'}}\\
+        \varUpdate{\var{favs}\setminus\var{cur}}\\
+        \varUpdate{\var{avs'}}\\
+      \end{array}
+      \right)
+    }
+  \end{equation}
+
+  \nextdef
+
+  \begin{equation}\label{eq:update-consensus}
+    \inference[AV-Update-Consensus]
+    {
+      \var{aup}\neq\emptyset
+      &
+      \dom{\var{aup}}\subseteq\dom{\var{dms}}
+      \\
+      \var{aup'}\leteq\var{aup_s}\unionoverrideRight\var{aup}
+      &
+      \var{fav}\leteq\fun{votedValue_{Applications}}~\var{aup'}
+      \\
+      \var{fav}\neq\Nothing
+      \\
+      \var{cur}\leteq\{\var{s}\mapsto\var{v}\in\var{favs} ~\mid~ s\geq\var{slot}\}
+      \\
+      \var{avs'}\leteq\fun{newAVs}~\var{avs}~\var{cur}
+      \\
+      s\leteq\var{slot}+\SlotsPrior
+    }
+    {
+      \begin{array}{l}
+        \var{slot}\\
+        \var{dms}\\
+      \end{array}
+      \vdash
+      \left(
+      \begin{array}{l}
+        \var{aup_s}\\
+        \var{favs}\\
+        \var{avs}\\
+      \end{array}
+      \right)
+      \trans{avup}{aup}
+      \left(
+      \begin{array}{l}
+        \varUpdate{\emptyset}\\
+        \varUpdate{(\var{favs}\setminus\var{cur})
+          \unionoverrideRight\{\var{s}\mapsto\var{fav}\}}\\
+        \varUpdate{\var{avs'}}\\
+      \end{array}
+      \right)
+    }
+  \end{equation}
+
+  \caption{Application Version Update inference rules}
+  \label{fig:rules:av-update}
+\end{figure}
+
+\clearpage
+
+\begin{figure}[htb]
+  \emph{Update environment}
+  \begin{equation*}
+    \UpdateEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{slot} & \Slot & \text{current slot}\\
+        \var{dms} & \VKeyGen\mapsto\VKey & \text{genesis key delegations} \\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Update states}
+  \begin{equation*}
+    \UpdateState =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{pup} & \PPUpdate & \text{protocol parameter proposals} \\
+        \var{aup} & \AVUpdate & \text{application versions proposals} \\
+        \var{favs} & \Slot\mapsto\Applications & \text{future application versions} \\
+        \var{avs} & \Applications & \text{current application versions} \\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Update transitions}
+  \begin{equation*}
+    \_ \vdash
+    \var{\_} \trans{up}{\_} \var{\_}
+    \subseteq \powerset (\UpdateEnv \times \UpdateState \times \Update \times \UpdateState)
+  \end{equation*}
+  %
+  \caption{Application Version Update transition-system types}
+  \label{fig:ts-types:update}
+\end{figure}
+
+\begin{figure}[htb]
+  \begin{equation}\label{eq:update}
+    \inference[Update]
+    {
+      (\var{pup},~\var{aup})\leteq\var{up}
```

Comment:
Here it seems we're forcing an update to contain parameters and software updates. Do we need this constraint?